### ISSUE TITLE
Directus SDK readItems Type error with typscript version 5.4.5

### ISSUE THREAD

**Issue by ElliotHYLee** [directus/directus#22328]
### Describe the Bug

```
const result = await client.request(readItems('collection_name', query_object));
```
The above call had no issue with the typescript  version 5.3.3 until I updated ts 5.4.5

readItems() expects number| symbol only for the collection name.

"Argument of type 'string' is not assignable to parameter of type 'number | symbol'.ts(2345)" failes the build with the "collection_name" argument.

![Screenshot from 2024-04-26 17-53-58](https://github.com/directus/directus/assets/7537835/e2654f2e-c851-46ad-8875-98f3e70a4d56)

Although the above example is on nextjs, it shows the same behavior with nodejs app for backend. 

so far, I  found ts 5.3.3 works.

### To Reproduce

typescript 5.3.3 works to get any items using directus SDK of any version. 

just updated to typescript 5.4.5 and  breaks all readItems that expects number | symbols.

### Directus Version

v10.10.4

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by R1NEJ0** [comment]
In your create directus you have to set the collections like this:

![image](https://github.com/directus/directus/assets/26463620/d1e1563a-3043-4208-85c8-c3f2aac28c9e)

You can use this script (it's not mine) :
https://www.izoukhai.com/blog/generate-typescript-definitions-from-directus

**Comment by hanneskuettner** [comment]
Thanks for opening this issue! However this is a duplicate of an existing issue <ISSUE NUMBER>, so I'll proceed to close this but if you think this is actually a different issue, then please do not hesitate to comment. The existing issue #22206 has been fixed by #22211 which will be available with the next release.
